docs/DeveloperGuide.md

```diff
@@ -1,34 +1,600 @@
+    * [Appendix B: User Stories](#appendix-b-user-stories)
+    * [Appendix C: Non-functional Requirements](#appendix-c-non-functional-requirements)
+    * [Appendix D: Glossary](#appendix-d-glossary)
+    * [Appendix E: Instructions for Manual Testing](#appendix-e-instructions-for-manual-testing)
 
-### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+## 1. Introduction
 
-## User Stories
+### 1.1. Purpose
+
+This document contains the architecture and software design specifications for the application HappyPills.
+
+### 1.2. Scope
+
+This guide is mainly for developers, designers and software engineers that are working on and using HappyPills.
+
+## 2. Setting Up
+
+1. Ensure that you have `Java 11` or later installed in your computer 
+2. Click [here](link to be added later?) to download the latest HappyPills JAR File
+
+## 3. Design 
+
+This section provides a high-level overview of HappyPills.
+
+### 3.1. Architecture
+
+![Architecture diagram](images/ArchitectureDiagram.PNG "Overview of the Application")
+
+The **Architecture diagram** above explains the high-level design of HappyPills.
+
+Main has one class called `HappyPills`.  
+It is responsible for:  
+- At app launch: Initializes the components in the correct sequence, and connects them up with each other.
+- At shut down: Shuts down the components.
+
+The rest of the App consists of four components.
+
+- `UI`: The UI of the App.
+
+- `Logic`: The command executor.
+
+- `Model`: Holds the data of the App in-memory.
+
+- `Storage`: Reads data from and writes data to the text file.
+
+
+The architecture of HappyPills is broken down into seven main classes:
+* `Ui`: This class handles the User Interface of the application.
+* `Parser`: This class handles the parsing and handling of user commands.
+* `Command`: This class handles all the commands the application has.
+* `Storage`: This class reads and writes data to and from text files for future use.
+* `Patient`: This class manages the data of data type Patient in memory. 
+* `Appointment`: This class manages the data of data type Appointment in memory.
+* `MedicalRecords`: This class manages the data of data type MedicalRecord in memory.
+
+### 3.2. TextUi Component
+
+The `TextUi` component: 
+* Executes user commands using the command component.
+* Listens for changes and outputs messages accordingly from the Command component.
+
+### 3.3. Logic Component
+
+![Logic Diagram](images/LogicDiagram.jpg)
+
+The diagram above shows the logic flow of HappyPills.
+
+1. `HappyPills` uses the `Parser` to parse the commands given by the user.
+2. This results in a `Command` object being returned which is then executed by `HappyPills`.
+3. The command executed will cause a change in the program (e.g. Patient added into program). 
+4. The result of the command is then passed back to the `TextUi` to display the message. 
+
+### 3.4. Model Component
+
+![Model Diagram](images/ModelDiagram.jpg)
```

Comment:
![image](https://user-images.githubusercontent.com/51941453/77998906-682e2580-7364-11ea-9b8c-29f97ba5d21a.png)

Should it be a composition rather than an aggregation, since when the patient list is destroyed, the person is destroyed as well? Vice versa for ApptList and PatientRecordList